mlrun/runtimes/base.py

```diff
@@ -811,6 +811,52 @@ def delete_runtime_object_resources(
             label_selector = object_label_selector
         self.delete_resources(db, db_session, label_selector, force, grace_period)
 
+    def monitor_run(
+        self,
+        db: DBInterface,
+        db_session: Session,
+        project: str,
+        run_uid: str,
+        interval: int = 5,
+        timeout: int = 60 * 60 * 24,
+    ):
+        logger.debug(
+            "Starting run monitor loop",
+            project=project,
+            run_uid=run_uid,
+            timeout=timeout,
+            interval=interval,
+        )
+        try:
+            mlrun.utils.helpers.retry_until_successful(
```

Comment:
I know it's good for now (it works), but I think the `retry_until_successful` pattern is not fitting to this kind of monitoring loop. Also I saw mlrun's retry_until_successful is synchronous.
This should be something like this: https://stackoverflow.com/questions/37512182/how-can-i-periodically-execute-a-function-with-asyncio (even though you're in a threadpool)

Reply:
I'm not sure what you mean by "not fitting to this kind of monitoring loop", why ?
exactly because I'm in a threadpool I can not use asyncio or do async stuff

Reply:
This call does not exist anymore (as part of moving to one loop to monitor all runs), it's now just 
`run_function_periodically(int(config.runs_monitoring_interval), _monitor_runs)` which I think addresses your comment